espnet/nets/chainer_backend/rnn/training.py

```diff
@@ -17,10 +17,14 @@
 from chainer.training.updaters.multiprocess_parallel_updater import gather_params
 from chainer.training.updaters.multiprocess_parallel_updater import scatter_grads
 
-from cupy.cuda import nccl
-
 import numpy as np
 
+try:
+    # Calls nccl when cupy is installed for CustomParallelUpdater (only for gpu support)
+    from cupy.cuda import nccl
+except Exception:
+    pass
+
```

Comment:
Why don't you move above part to `CustomParallelUpdater.__init__()`?

Reply:
Just call `from cupy.cuda import nccl` in `CustomParallelUpdater.__init__()`.

Reply:
I agree. Because nccl is only used for multi GPU training, it should be also optional for single GPU training.